pkg/sidecar/sidecar.go

```diff
@@ -183,6 +232,131 @@ func (s *Sidecar) dumpBundles(svidResponse *workloadapi.X509Context) error {
 	return nil
 }
 
+func (s *Sidecar) writeJSON(fileName string, certs map[string]interface{}) {
+	file, err := json.Marshal(certs)
+	if err != nil {
+		s.config.Log.Errorf("Unable to parse certs: %v", err)
+	}
+
+	jsonPath := path.Join(s.config.CertDir, fileName)
+	if err = os.WriteFile(jsonPath, file, os.ModePerm); err != nil {
+		s.config.Log.Errorf("Unable to write JSON file: %v", err)
+	}
+}
+
+func (s *Sidecar) updateJWTBundle(jwkSet *jwtbundle.Set) {
+	s.config.Log.Debug("Updating JWT bundle")
+
+	bundles := make(map[string]interface{})
+	for _, bundle := range jwkSet.Bundles() {
+		bytes, err := bundle.Marshal()
+		if err != nil {
+			s.config.Log.Errorf("Unable to marshal JWT bundle: %v", err)
+			continue
+		}
+		bundles[bundle.TrustDomain().Name()] = base64.StdEncoding.EncodeToString(bytes)
+	}
+
+	s.writeJSON(s.config.JWTBundleFilename, bundles)
+	s.config.Log.Info("JWT bundle updated")
+}
+
+func (s *Sidecar) fetchJWTSVID(options ...workloadapi.ClientOption) (*jwtsvid.SVID, error) {
+	clientOptions := workloadapi.WithClientOptions(options...)
+
+	jwtSource, err := workloadapi.NewJWTSource(context.Background(), clientOptions)
```

Comment:
this source can be reused, actual code will fetch bundles multiple times, 
ctx must be passed from caller to be attached to service, this source can prevent things to stop

Reply:
@JU4N98 This comment has not been addressed yet.